### PR TITLE
Add jsdoc and update general documentation

### PR DESCRIPTION
**Pull request by jlks**
📺 What

This adds jsdoc comments around `bigscreenplayer.js` and generates static HTML documentation to the `/docs` folder.

🛠 How

- Integrate jsdoc with the docdash template
- Improve the `bigscreenplayer.js` api documentation.
- Should be able to point to this over the wiki API page that is out of date.

### CODE REVIEW

**Comment by ShiningTrapez** [comment]
Do we need every variant of every font?

File: script/bigscreenplayer.js [hunk 1]
```diff
@@ -255,75 +374,209 @@ define('bigscreenplayer/bigscreenplayer',
...
+         * @function
+         * @return if subtitles are currently enabled.
+         */
         isSubtitlesEnabled: function () {
           return playerComponent ? playerComponent.isSubtitlesEnabled() : false;
         },
+
+        /**
+         * @function
+         * @return Returns whether or no subtiles are currently enabled.
```
Comment (matt-stephenson) [inline script/bigscreenplayer.js:444]: Typo in 'subtitles' and also should be 'not'

File: script/models/windowtypes.js [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+ */
 define(
   'bigscreenplayer/models/windowtypes',
   function () {
     'use strict';
-    return {
+    return /** @alias module:bigscreenplayer/models/windowtypes */ {
+      /** Media with a duration */
       STATIC: 'staticWindow',
+      /** Media with a start without an initial duration */
```
Comment (matt-stephenson) [inline script/models/windowtypes.js:13]: Reword as 'start time but without...' maybe? The sentence structure just confused me :)

**Comment by domjacks** [comment]
This was completed as part of #246

**Pull request closed without merging**